src/libpmemobj/tx.c

```diff
@@ -820,16 +829,21 @@ pmemobj_tx_begin(PMEMobjpool *pop, jmp_buf env, ...)
 int
 pmemobj_tx_xlock(enum pobj_tx_param type, void *lockp, uint64_t flags)
 {
+	struct tx *tx = get_tx();
+	ASSERT_IN_TX(tx);
+	ASSERT_TX_STAGE_WORK(tx);
+
+	struct tx_data *txd = PMDK_SLIST_FIRST(&tx->tx_entries);
```

Comment:
this pattern is repeated multiple times, wrap it into a function:
`flags |= tx_abort_flag(tx);`